### PR TITLE
fix: remove incorrect stack transform from charts with one linear Q-axis one non-linear Q-axis.

### PR DESCRIPTION
**Pull request by kanitw**
Remove incorrect stack transform from charts with one linear Q-axis one non-linear Q-axis.

### CODE REVIEW

File: src/stack.ts [hunk 1]
```diff
@@ -104,15 +104,6 @@ function potentialStackedChannel(
       // if there is no explicit stacking, only apply stack if there is only one aggregate for x or y
       if (xAggregate !== yAggregate) {
         return xAggregate ? x : y;
-      } else {
```
Comment (kanitw) [inline src/stack.ts:107]: This logic was just wrong. Instead, we should rely on the orient logic below.

**Review by ZacharyBys — APPROVED** [review]

File: examples/compiled/bar_q_qpow.png
Comment (ZacharyBys) [inline examples/compiled/bar_q_qpow.png:1]: Are these two images supposed to be before and after? I don't see any changes

**Review by domoritz — APPROVED** [review]
Looks good. There is a failing test for bar_q_qpow.vl.json, though.

File: examples/compiled/bar_q_qpow.png
Comment (kanitw) [reply examples/compiled/bar_q_qpow.png:1]: If you see the 4th (generated commits), you can see that the actual code changes in the 3rd commit fixes the problem.

Comment (kanitw) [reply examples/compiled/bar_q_qpow.png:1]: image

**Pull request merged**